### PR TITLE
ENH: Add georeference module, basic tests

### PR DESCRIPTION
**Pull request by mgrover1**
Add some georeference helper functions, first steps to georeferencing the xradar object

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
Thanks @mgrover for setting this up. :+1: 

Just some questions, comments and thoughts:

- Is there a specific need to have `geographic_to_cartesian_aeqd` in the codebase?
- It would be great to parameterize earth radius as well as k.
- I'd suggest to use metric SI units (meters in this example), as the model keeps the ranges in meters.
- We can use site-coordinates (lat, lon, alt) to calculate final bin aeqd coordinates.
- We might use latitude dependend earth radius for the calculations.
- We might return a projstring with the used AEQD projection.

Basically in wradlib we are using the same equations derived Doviak. We might utilize the existing wradlib function which adds the x, y, z and groundrange coordinate DataArrays.

**Comment by mgrover1** [comment]
Yeah - on second thought, I like that idea of operating at the **dataset level** instead of the numpy level. This is using the current functions/methods we have in Py-ART, but adapting it to work at the dataset level is a better approach.

I will make those changes, and ping you when it is ready for review.

File: xradar/georeference/transforms.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    elevation angle, s is the arc length, and R is the effective radius
+    of the earth, taken to be 4/3 the mean radius of earth (6371 km).
+    References
+    ----------
+    .. [1] Doviak and Zrnic, Doppler Radar and Weather Observations, Second
+        Edition, 1993, p. 21.
+    """
+    if effective_radius_fraction is None:
+        effective_radius_fraction = 4.0 / 3.0
+    theta_e = np.deg2rad(elevations)  # elevation angle in radians.
```
Comment (kmuehlbauer) [inline xradar/georeference/transforms.py:50]: Just a personal style thing, I'd prefer comments above. But no strict necessity.

File: xradar/georeference/transforms.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    ds['x'], ds['y'], ds['z'] = antenna_to_cartesian(
+        ds.range,
+        ds.azimuth,
+        ds.elevation,
+        earth_radius=earth_radius,
+        effective_radius_fraction=None,
+    )
+
+    # Set the attributes for the dataset
+    ds.x.attrs = {'standard_name': 'east_west_distance_from_radar', 'units': 'meters'}
```
Comment (kmuehlbauer) [inline xradar/georeference/transforms.py:107]: Let's move those attributes into the model.

File: examples/notebooks/plot-ppi.ipynb [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eaa2ee4c-98f2-41d2-b16e-9d05ad8608da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = xd.georeference.get_x_y_z(ds)\n",
```
Comment (kmuehlbauer) [inline examples/notebooks/plot-ppi.ipynb:98]: Maybe also advertise the `.pipe()`

`ds = ds.pipe(georeference.get_x_y_z)`

File: tests/georeference/test_transforms.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    assert_almost_equal(y[np.where(azimuths == 270.0)], 0)
+
+
+def test_get_x_y_z():
+
+    # Create an empty xradar dataset
+    ds = xradar.model.Dataset()
+
+    # Add similar data to before
+    ds['time'] = pd.date_range(start='2018-11-10', end='2018-11-10T00:09:00', freq='T')
```
Comment (kmuehlbauer) [inline tests/georeference/test_transforms.py:35]: We could facilitate `xradar.model.create_sweep_dataset` here.

**Review by kmuehlbauer — APPROVED** [review]
@mgrover1 This looks very neat throughout. Just a couple of minor comments and suggestions. Those might be addressed here on a follow up PR.

I've fixed a bit for the docs, to be able to see it in all it's glory.

File: tests/georeference/test_transforms.py (hunk 4 above)
Comment (kmuehlbauer) [reply tests/georeference/test_transforms.py:35]: Double take here

File: tests/georeference/test_transforms.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    assert_almost_equal(x[np.where(azimuths == 180.0)], 0)
+
+    # Make sure that at 270 degrees, y is (close to) 0
+    assert_almost_equal(y[np.where(azimuths == 270.0)], 0)
+
+
+def test_get_x_y_z():
+
+    # Create an empty xradar dataset
+    ds = xradar.model.Dataset()
```
[the remark this answers is not in the document]
Comment (mgrover1) [reply tests/georeference/test_transforms.py:32]: I like these approaches! Let's add these to the examples too.

File: xradar/georeference/transforms.py (hunk 1 above)
Comment (mgrover1) [reply xradar/georeference/transforms.py:50]: Me too - I'll correct that

File: examples/notebooks/plot-ppi.ipynb (hunk 3 above)
Comment (mgrover1) [reply examples/notebooks/plot-ppi.ipynb:98]: Yeah - that might be good - I think this would be good in a follow up PR

File: xradar/georeference/transforms.py (hunk 2 above)
Comment (mgrover1) [reply xradar/georeference/transforms.py:107]: > @mgrover1 This looks very neat throughout. Just a couple of minor comments and suggestions. Those might be addressed here on a follow up PR.

Can we address these with a follow-up PR?

**Comment by mgrover1** [comment]
> 
Let's address these with another PR.

**Comment by kmuehlbauer** [comment]
Ready to go, @mgrover1 ?

**Comment by mgrover1** [comment]
Sure :)

**Comment by kmuehlbauer** [comment]
Thanks @mgrover1, this is going to be fun to add to the current examples.
